src/App.tsx

```diff
@@ -26,70 +27,126 @@ type ReorderOptions = {
+  ) => {
+    const visibleGoods = [...goods];
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+    if (sortType === 1) {
+      visibleGoods.sort((a, b) => {
+        return a.localeCompare(b);
+      });
+    }
+
+    if (sortType === 2) {
+      visibleGoods.sort((a, b) => {
+        return a.length - b.length;
+      });
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  sortAlphabetically = () => {
+    this.setState(
+      { sortType: SortType.ALPHABET },
+    );
+  };
+
+  sortLength = () => {
+    this.setState(
+      { sortType: SortType.LENGTH },
+    );
+  };
+
+  reset = () => {
+    this.setState(
+      {
+        isReversed: false,
+        sortType: SortType.NONE,
+      },
+    );
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  render() {
+    const reorderedGoods = this.getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${this.state.sortType !== 1 ? 'is-light' : ''}`}
+            onClick={this.sortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success ${this.state.sortType !== 2 ? 'is-light' : ''}`}
+            onClick={this.sortLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning ${!this.state.isReversed ? 'is-light' : ''}`}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+          {
+            JSON.stringify(reorderedGoods)
+              !== JSON.stringify(goodsFromServer)
+              ? (
+                <button
+                  type="button"
+                  className="button is-danger is-light"
+                  onClick={this.reset}
+                >
+                  Reset
+                </button>
+              ) : <></>
```

Comment:
No need to return an empty tag. Just use && like this:
`true && <button>...</button>`